src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationExtensions.cs

```diff
@@ -78,7 +73,37 @@ public static IConfigurationBuilder SetBasePath(this IConfigurationBuilder build
                 throw new ArgumentNullException(nameof(basePath));
             }
 
+            builder.Properties[BasePathKey] = basePath;
```

Comment:
We should revisit what UserSecrets is doing, as far as I can tell, we always need to set the file provider anyways for user secrets to work (since the secrets file lives outside of the normal app base path), so we might be able to just remove all the seemingly unneeded calls to GetBasePath from UserSecrets.  There is some really weird code over there where its calling GetBasePath several times in each extension method.